src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -52,109 +64,62 @@
  */
 public class HTTPRecordWriter extends RecordWriter<StructuredRecord, StructuredRecord> {
   private static final Logger LOG = LoggerFactory.getLogger(HTTPRecordWriter.class);
+  private static final String REGEX_HASHED_VAR = "#s*(\\w+)";
+
   private final HTTPSinkConfig config;
   private final MessageBuffer messageBuffer;
   private String contentType;
+  private String url;
+  private String configURL;
+  private List<PlaceholderBean> placeHolderList;
+
   private AccessToken accessToken;
+  private final HttpErrorHandler httpErrorHandler;
+  private final PollInterval pollInterval;
+  private int httpStatusCode;
 
   HTTPRecordWriter(HTTPSinkConfig config, Schema inputSchema) {
     this.config = config;
     this.accessToken = null;
     this.messageBuffer = new MessageBuffer(
-            config.getMessageFormat(), config.getJsonBatchKey(), config.shouldWriteJsonAsArray(),
-            config.getDelimiterForMessages(), config.getCharset(), config.getBody(), inputSchema
+      config.getMessageFormat(), config.getJsonBatchKey(), config.shouldWriteJsonAsArray(),
+      config.getDelimiterForMessages(), config.getCharset(), config.getBody(), inputSchema
     );
+    this.httpErrorHandler = new HttpErrorHandler(config);
+    if (config.getRetryPolicy().equals(RetryPolicy.LINEAR)) {
+      pollInterval = FixedPollInterval.fixed(config.getLinearRetryInterval(), TimeUnit.SECONDS);
+    } else {
+      pollInterval = IterativePollInterval.iterative(duration -> duration.multiply(2),
+                                                     Duration.FIVE_HUNDRED_MILLISECONDS);
+    }
+    url = config.getUrl();
+    placeHolderList = getPlaceholderListFromURL();
   }
 
   @Override
   public void write(StructuredRecord input, StructuredRecord unused) throws IOException {
+    configURL = url;
     if (config.getMethod().equals("POST") || config.getMethod().equals("PUT")) {
       messageBuffer.add(input);
     }
     if (config.getBatchSize() == messageBuffer.size()) {
```

Comment:
This should be removed or this can cause duplicate. 

This will send message before replacing the placeholder. 
And later it will replace and send the message again.  (line 112)

Reply:
I think this got introduced twice after rebasing with the `develop` branch. Removed the duplicate `if` block dde1cd273c0c4fee6e7c70505a080df0b47f24f6